design.md

```diff
@@ -39,16 +48,43 @@
   - Kinematics (e.g., bases, limbs, links, DH parameters, root offsets).
   - Scripts (e.g., body/limb/joint controllers, body/limb CAD generators).
 
+### Plugins
+
+- The IDE must be able to display a searchable list of plugins.
+  - The IDE must display the group, name, and version triple.
+  - The IDe must display the plugin artifact's description.
+  - By default, the IDE displays all plugins in the group `com.commonwealthrobotics`.
+  - The user may filter for a specific type of plugin (e.g. device, toolchain, device resource, or simulator).
+- There is a list of default plugins which are installed by default.
+
+### Git and GitHub Integration
+
+- The IDE must present a simplified view of Git and GitHub to the user.
+  - The user must be able to commit and push their changes from one simplified modal interface.
+
 ### Interface with the Kernel
 
 - The kernel daemon must be started and connected to (if not already connected) when the user loads a project.
+  - When the kernel is connected, the plugin should load the display daemon and connect the kernel to it.
 - If the kernel daemon becomes unresponsive, it must be restarted.
-- If the project is closed, the kernel daemon must be killed.
+- If the project is closed, the kernel and display daemons must be killed.
   - There must be a graceful shutdown method that is tried first. If that fails, the process must be killed by the OS.
+- If the kernel needs credentials, the IDE automatically gives the user's credentials.
+  - The user can disable handling this automatically.
```

Comment:
Do we want to specify specifically what options the user has? I'm imagining "remember this?" "yes"/"no"/"only for this session" or something like that.

Reply:
Yes.